src/ApiConnector.php

```diff
@@ -242,4 +249,25 @@ private static function sanitizeData($data): ?string
 
         return $data;
     }
+
+    public static function initializeOAuth(OAuth $oauth, string $apiVersion = 'v3')
+    {
+        $handler = HandlerStack::create();
+        $handler->push($oauth->getMiddleware());
+
+        $hostname = $oauth->getEnvironment() === 'sandbox' ? 'sandbox.penneo.com' : 'app.penneo.com';
```

Comment:
Since the environment is already fixed in the passed config, why hardcoding sandbox and app here? If it's because of the mapping "production -> app", then I think the mapping should be done elsewhere.

Reply:
After reading the second comment on the topic, clearly I have not understood. Could you provide an example for the mapper you have in mind?

Reply:
I've moved the logic into the `Envioronment` helper class.